x-pack/plugins/security_solution/public/management/pages/event_filters/view/components/form.tsx

```diff
@@ -411,6 +437,122 @@ export const EventFiltersForm: React.FC<ArtifactFormComponentProps & { allowSele
       [nameInputMemo, descriptionInputMemo]
     );
 
+    const handleFilterTypeOnChange = useCallback(
+      (id: string) => {
+        const newTagsForDescendents =
+          id === 'descendents' ? [FILTER_DESCENDENTS_OF_PROCESS_TAG] : [];
+
+        const tags = getTagsUpdatedBy('descendentOfProcessFiltering', newTagsForDescendents);
+
+        processChanged({ tags });
+        if (!hasFormChanged) setHasFormChanged(true);
+      },
+      [getTagsUpdatedBy, hasFormChanged, processChanged]
+    );
+
+    const filterTypeOptions: EuiButtonGroupOptionProps[] = useMemo(() => {
+      const descendentsTooltip = (
+        <EuiToolTip
+          content={
+            <FormattedMessage
+              id="xpack.securitySolution.eventFilters.filterDescendentsOfProcess.tooltip"
+              defaultMessage="Filtering the descendents of a process means that events from the matched process are still ingested, but the ones from its descendent processes will be omitted."
+            />
+          }
+          data-test-subj={getTestId('filterProcessDescendents-tooltipText')}
+        >
+          <EuiIcon
+            type="iInCircle"
+            data-test-subj={getTestId('filterProcessDescendents-tooltipIcon')}
+          />
+        </EuiToolTip>
+      );
+
+      return [
+        {
+          id: 'events',
+          label: (
+            <EuiText size="s">
+              <FormattedMessage
+                id="xpack.securitySolution.eventFilters.filterDescendentsOfProcess.eventsButton"
+                defaultMessage="Events"
+              />
+            </EuiText>
+          ),
+          iconType: isFilterDescendentsOfProcessSelected ? 'empty' : 'checkInCircleFilled',
+          'data-test-subj': getTestId('filterEventsButton'),
+        },
+        {
+          id: 'descendents',
+          label: (
+            <EuiFlexGroup direction="row" gutterSize="s" alignItems="center">
+              <EuiText size="s">
+                <FormattedMessage
+                  id="xpack.securitySolution.eventFilters.filterDescendentsOfProcess.descendentsOfProcessButton"
+                  defaultMessage="Descendents of process"
+                />
+              </EuiText>
+              {descendentsTooltip}
+            </EuiFlexGroup>
+          ),
+          iconType: isFilterDescendentsOfProcessSelected ? 'checkInCircleFilled' : 'empty',
+          'data-test-subj': getTestId('filterProcessDescendentsButton'),
+        },
+      ];
+    }, [getTestId, isFilterDescendentsOfProcessSelected]);
+
+    const filterTypeSubsection = useMemo(() => {
+      if (!isFilterDescendentsOfProcessFeatureEnabled) return null;
+
+      return (
+        <>
+          <EuiButtonGroup
+            legend="Events or Process descendents selector"
+            color="primary"
+            onChange={handleFilterTypeOnChange}
+            css={css`
+              .euiButtonGroupButton {
+                padding-right: ${euiTheme.euiTheme.size.l};
```

Comment:
Did the design specifically asked to make it large?  If so, just ensure the UX Designer know that the current EUI Button Group only supports (va the `buttonSize` prop) `s', `m` and `compressed` and confirm that they still want this change to be applied

Reply:
there is no design, here I reused the button group from the Global/PerPolicy policy selector on the same flyout:

https://github.com/elastic/kibana/blob/9984ce88a6e6fedc68c65628bc16ab046294bab2/x-pack/plugins/security_solution/public/management/components/effected_policy_select/effected_policy_select.tsx#L55-L61

do you think we are good with this?

Reply:
Ok. thanks. I just want to ensure every time I see changes to the out-of-the-box EUI styles that we question their need so that we don't incur increase maintenance of our components when EUI changes.